javacord-api/src/main/java/org/javacord/api/entity/VanityUrlCode.java

```diff
@@ -0,0 +1,40 @@
+package org.javacord.api.entity;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class VanityUrlCode {
+
+    private final String code;
+
+    /**
+     * Creates a new VanityUrlCode.
+     *
+     * @param vanityUrlCode The vanity code
+     */
+    public VanityUrlCode(String vanityUrlCode) {
+        code = vanityUrlCode;
+    }
+
+    /**
+     * Gets the vanity code.
+     *
+     * @return The vanity code.
+     */
+    public String getCode() {
+        return code;
+    }
+
+    /**
+     * Gets the vanity url.
+     *
+     * @return The URL of the vanity code.
+     */
+    public URL getUrl() {
+        try {
+            return new URL("https://discordapp.com/invite/" + code);
+        } catch (MalformedURLException e) {
+            return null;
```

Comment:
I know we have this in other places like this, but it would be better to have it like this.
```suggestion
            throw new AssertionError("Unexpected malformed vanity url", e);
```